### PR TITLE
fix comparison between multiple patterns issue

### PR DESCRIPTION
**Pull request by daejunpark**
when multiple patterns exist, comparison should be made with considering their klabels as well as their inputs.

@andreistefanescu please review

### CODE REVIEW

**Comment by andreistefanescu** [comment]
ups... I forgot about that

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicUnifier.java [hunk 1]
```diff
@@ -388,7 +388,8 @@ private void unifyMap(BuiltinMap map, BuiltinMap otherMap) {
         List<KItem> otherRemainingPatterns = new ArrayList<>();
         for (KItem pattern : patterns) {
             for (KItem otherPattern : otherPatterns) {
-                if (pattern.getPatternInput().equals(otherPattern.getPatternInput())) {
+                if (pattern.getPatternInput().equals(otherPattern.getPatternInput())
```
Comment (andreistefanescu) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicUnifier.java]: I would do the `KLabel` check first

**Comment by daejunpark** [comment]
@andreistefanescu please review again

**Comment by andreistefanescu** [comment]
lgtm

**Pull request merged**